### ISSUE TITLE
object or collection level storage management

### ISSUE THREAD

**Issue by drheme** [isar/isar#387]
**Is your feature request related to a problem? Please describe.**

Our database contains much objects that become obsolete after some time and hope to clean it up when it's taking too much storage space.

**Describe the solution you'd like**

a) We plan to trigger cleanup of our database when the storage reaches certain threshold.
b) The preferred way is to automatically remove the least recently accessed objects until a certain storage usage bound.

This might be provided via schema annotation or database options.

**Describe alternatives you've considered**

There are different alternatives, one proximation is that we can do storage accounting within an object.
And query for total storage and delete by access time.

```
@Collection()
class BigObject {
  int? id;

  DateTime lastAccess;

  int objectSize;
}
```

**Version**
 - Platform: **iOS, Android**, Mac, Windows, Linux, Web
 - Flutter version: [e.g. 1.5.4] latest
 - Isar version: [e.g. 0.5.0] latest

**Comment by richard457** [comment]
This could be a good feature, especially on mobile devices where storage is limited, however, I am thinking of this feature this way:
- What will happen when sync is introduced
-  if local objects are deleted what will happen to remote objects?

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.2`
